client-extensions/liferay-object-collection-filter-extension/src/common/components/FilterSettings.js

```diff
@@ -1,24 +1,27 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
 import React, { useState, useEffect } from 'react';
-import Select from 'react-select';
+import ClayButton from '@clayui/button';
+import ClayModal from '@clayui/modal';
+import { useModal } from '@clayui/modal';
+import ClayLayout from '@clayui/layout';
+import ObjectChosenDropButton from './SelectObjectDefinition';
+import FieldChosenDropButton from './SelectObjectFields';
 
 const FilterSettings = ({ initialSettings, onSaveAndClose }) => {
 
     const [settings, setSettings] = useState(initialSettings);
 
-    const [objects, setObjects] = useState([]);
-    const [objectFields, setObjectFields] = useState([]);
-
-    useEffect(() => {
-        // Simulate fetching data (replace with actual fetching logic)
-        setObjects([
-           { label: 'students', value: 'students'},
-           { label: 'teachers', value: 'teachers'},
-       ]);
-        setObjectFields([
-            { label: 'age', fieldName: 'age', fieldType: 'numeric' },
-            { label: 'gender', fieldName: 'gender', fieldType: 'boolean' },
-        ]);
-    }, []);
+    const { observer, onOpenChange, open } = useModal();
+    const [selectedData, setSelectedData] = React.useState(null);
```

Comment:
you already import useState , should use const [selectedData, setSelectedData] = useState(null);